Lesson6/hw_ig.py

```diff
@@ -0,0 +1,55 @@
+# 1. Implement a function that flatten incoming data:
+# non-iterables and elements from iterables (any nesting depth should be supported)
+# function should return an iterator (generator function)
+# don't use third-party libraries
+
+
+def merge_elems(*elems):
+    for iterable in elems:
+        if isinstance(iterable, (list, tuple)):
+            yield from merge_elems(*iterable)
+        elif not hasattr(iterable, '__iter__'):
+            yield iterable
+        else:
+            for item in iterable:
+                yield item
+
+
+# example input
+a = [1, 2, 3]
+b = 6
+c = 'zhaba'
+d = [[1, 2], [3, 4]]
+
+for _ in merge_elems(a, b, c, d):
```

Comment:
input: a = e = {'aaa': 22, 'bbb': 66}
expected_result: a a a  b b b (dictionary is iterable object, keys are also iterable)
actual_result: aaa bbb

Reply:
I've added dict to the condition check. I'm not sure if it's a universal solution, but it works)

Reply:
Have changed. Do you mean this way?